Flaky e2e: test_axon/test_axon_async assert against a second, independent get_external_ip() lookup

## Summary

`tests/e2e_tests/test_axon.py::test_axon` and `::test_axon_async` are flaky on CI runners that have more than one outbound/egress IP. Each test serves an axon (whose IP is resolved via `bittensor.utils.networking.get_external_ip()`), then calls `get_external_ip()` **a second time** to compute the "expected" value and asserts the two are equal. Since `get_external_ip()` is a live network lookup, the two independent calls can return different addresses, so the assertion fails intermittently.

## Where

`tests/e2e_tests/test_axon.py` (`staging`):
- `test_axon`: `external_ip = networking.get_external_ip()` (~L55), then `assert updated_axon.ip == external_ip` (~L66)
- `test_axon_async`: `external_ip = networking.get_external_ip()` (~L127), then `assert updated_axon.ip == external_ip` (~L138)

`updated_axon.ip` is itself resolved by `get_external_ip()` inside the miner template's `serve`, so the assertion effectively compares **two independent `get_external_ip()` results**.

`bittensor/utils/networking.py::get_external_ip()` resolves the IP from external services in sequence:
- `https://checkip.amazonaws.com`
- `https://ident.me`
- `https://www.wikipedia.org` (`X-Client-IP` header)

## Observed failure

On a self-hosted Azure runner with an egress NAT pool in `48.211.213.x`:

```
>   assert updated_axon.ip == external_ip
E   AssertionError: Expected IP 48.211.213.35, but got 48.211.213.36
E   assert '48.211.213.36' == '48.211.213.35'
tests/e2e_tests/test_axon.py:138: AssertionError
```

The serve resolved `.36`; the test's follow-up lookup resolved `.35`. Both are valid egress IPs of the same NAT pool — the two lookups simply left via different addresses (and the rotating list of services can also report different ones). It reproduced across retries on the same runner ("failed after 2 attempts").

## Why

- CI runners use an outbound NAT pool / load balancer with multiple public IPs, so two outbound requests can egress from different addresses.
- `get_external_ip()` is non-deterministic (live lookup with multiple fallback services), so calling it twice and asserting equality isn't safe.

## Proposed fix

Resolve the external IP **once** and compare the served axon against that single value, instead of a second fresh lookup. Options:
- Capture `external_ip` and pass it into the miner/serve so the on-chain IP and the expected value share one source of truth, then assert against it.
- Or assert the served IP is *a* valid public IP (and `ip_type` matches) rather than exact-matching a fresh lookup.
- Or mock/cache `get_external_ip()` for the test's duration.

This is purely a test-hygiene issue — it surfaced while running these e2e tests against `subtensor` PRs (the underlying code under test is fine).

--- a/tests/e2e_tests/test_axon.py
+++ b/tests/e2e_tests/test_axon.py
@@ -1,6 +1,7 @@
 import asyncio
 import time
 
+import netaddr
 import pytest
 
 from bittensor.utils import networking
@@ -52,7 +53,6 @@ def test_axon(subtensor, templates, alice_wallet):
         # Refresh the metagraph
         metagraph = subtensor.metagraphs.metagraph(alice_sn.netuid)
         updated_axon = metagraph.axons[0]
-        external_ip = networking.get_external_ip()
 
     # Assert updated attributes
     assert len(metagraph.axons) == 1, (
@@ -63,13 +63,9 @@ def test_axon(subtensor, templates, alice_wallet):
         f"Expected 1 neuron, but got {len(metagraph.neurons)}"
     )
 
-    assert updated_axon.ip == external_ip, (
-        f"Expected IP {external_ip}, but got {updated_axon.ip}"
-    )
-
-    assert updated_axon.ip_type == networking.ip_version(external_ip), (
-        f"Expected IP type {networking.ip_version(external_ip)}, but got {updated_axon.ip_type}"
-    )
+    assert updated_axon.ip != "0.0.0.0", f"Axon IP was not updated: {updated_axon.ip}"
+    assert netaddr.IPAddress(updated_axon.ip).is_global()
+    assert updated_axon.ip_type == networking.ip_version(updated_axon.ip)
 
     assert updated_axon.port == 8091, f"Expected port 8091, but got {updated_axon.port}"
 
@@ -124,7 +120,6 @@ async def test_axon_async(async_subtensor, templates, alice_wallet):
         # Refresh the metagraph
         metagraph = await async_subtensor.metagraphs.metagraph(alice_sn.netuid)
         updated_axon = metagraph.axons[0]
-        external_ip = networking.get_external_ip()
 
     # Assert updated attributes
     assert len(metagraph.axons) == 1, (
@@ -135,13 +130,9 @@ async def test_axon_async(async_subtensor, templates, alice_wallet):
         f"Expected 1 neuron, but got {len(metagraph.neurons)}"
     )
 
-    assert updated_axon.ip == external_ip, (
-        f"Expected IP {external_ip}, but got {updated_axon.ip}"
-    )
-
-    assert updated_axon.ip_type == networking.ip_version(external_ip), (
-        f"Expected IP type {networking.ip_version(external_ip)}, but got {updated_axon.ip_type}"
-    )
+    assert updated_axon.ip != "0.0.0.0", f"Axon IP was not updated: {updated_axon.ip}"
+    assert netaddr.IPAddress(updated_axon.ip).is_global()
+    assert updated_axon.ip_type == networking.ip_version(updated_axon.ip)
 
     assert updated_axon.port == 8091, f"Expected port 8091, but got {updated_axon.port}"
 

